js.go

```diff
@@ -1068,6 +1106,85 @@ func (js *js) AddConsumer(stream string, cfg *ConsumerConfig) (*ConsumerInfo, er
 	return info.ConsumerInfo, nil
 }
 
+// JSApiConsumerDeleteResponse.
+type JSApiConsumerDeleteResponse struct {
+	APIResponse
+	Success bool `json:"success,omitempty"`
+}
+
+func (js *js) DeleteConsumer(stream, durable string) error {
+	if stream == _EMPTY_ {
+		return ErrStreamNameRequired
+	}
+
+	dcSubj := js.apiSubj(fmt.Sprintf(JSApiConsumerDeleteT, stream, durable))
+	r, err := js.nc.Request(dcSubj, nil, js.wait)
+	if err != nil {
+		return err
+	}
+	var resp JSApiConsumerDeleteResponse
+	if err := json.Unmarshal(r.Data, &resp); err != nil {
+		return err
+	}
+	if resp.Error != nil {
+		return errors.New(resp.Error.Description)
+	}
+	return nil
+}
+
+func (js *js) ConsumerInfo(stream, durable string) (*ConsumerInfo, error) {
+	return js.getConsumerInfo(stream, durable)
+}
+
+// ApiPagedRequest includes parameters allowing specific pages to be requests from APIs responding with ApiPaged
+type ApiPagedRequest struct {
+	Offset int `json:"offset"`
+}
+
+type JSApiConsumersRequest struct {
+	ApiPagedRequest
+}
+
+// ApiPaged includes variables used to create paged responses from the JSON API
+type ApiPaged struct {
+	Total  int `json:"total"`
+	Offset int `json:"offset"`
+	Limit  int `json:"limit"`
+}
+
+// JSApiConsumerListResponse.
+type JSApiConsumerListResponse struct {
+	APIResponse
+	ApiPaged
+	Consumers []*ConsumerInfo `json:"consumers"`
+}
+
+func (js *js) ListConsumers(stream string, offset int) ([]*ConsumerInfo, error) {
```

Comment:
Be good if we could hide the paging at this layer or undo it below the covers and just make it a stream of messages with EOF.

Reply:
All right. I added a `StreamLister` and `ConsumerLister` to iterate through pages.

This is what the calling code looks like.
```go
sl := js.NewStreamLister()
for sl.Next() {
	// sl.Page() returns a []*StreamInfo.
	// Each iteration can return a maximum of 256 items.
	fmt.Println(sl.Page())
}
if err := sl.Err(); err != nil {
	return err
}
```

Here's what it looks like in tests.
https://github.com/nats-io/nats.go/pull/626/files#diff-9e6bdee4051a6e36d6d507dcef63b5f9a39b281e421c08f02ec74a0fb6639e3aR592-R626